src/System.CommandLine/Parsing/ArgumentResult.cs

```diff
@@ -80,90 +80,66 @@ public void OnlyTake(int numberOfTokens)
         /// <inheritdoc/>
         public override string ToString() => $"{GetType().Name} {Argument.Name}: {string.Join(" ", Tokens.Select(t => $"<{t.Value}>"))}";
 
-        internal ParseError? CustomError(Argument argument)
+        /// <inheritdoc/>
+        public override void ReportError(string errorMessage)
         {
-            for (var i = 0; i < argument.Validators.Count; i++)
-            {
-                var validateSymbolResult = argument.Validators[i];
-                validateSymbolResult(this);
-
-                if (!string.IsNullOrWhiteSpace(ErrorMessage))
-                {
-                    return new ParseError(ErrorMessage!, Parent is OptionResult option ? option : this);
-                }
-            }
-
-            return null;
+            SymbolResultTree.ReportError(new ParseError(errorMessage, Parent is OptionResult option ? option : this));
+            _conversionResult = ArgumentConversionResult.Failure(this, errorMessage, ArgumentConversionResultType.Failed);
```

Comment:
at this moment we know that it's a failure

Reply:
Very nice. As long as we don't later allow clearing of the errors.